cmd/clusterctl/cmd/root.go

```diff
@@ -56,16 +56,6 @@ var RootCmd = &cobra.Command{
 	Long: LongDesc(`
 		Get started with Cluster API using clusterctl to create a management cluster,
 		install providers, and create templates for your workload cluster.`),
-	PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
-		// Check if Config folder (~/.cluster-api) exist and if not create it
-		configFolderPath := filepath.Join(homedir.HomeDir(), config.ConfigFolder)
-		if _, err := os.Stat(configFolderPath); os.IsNotExist(err) {
-			if err := os.MkdirAll(filepath.Dir(configFolderPath), os.ModePerm); err != nil {
-				return errors.Wrapf(err, "failed to create the clusterctl config directory: %s", configFolderPath)
-			}
-		}
-		return nil
-	},
 	PersistentPostRunE: func(cmd *cobra.Command, args []string) error {
```

Comment:
Is this behaviour retained in `xdg.ConfigFile`? It seems so but just wnat to make sure.

Reply:
Yes, the library will create the folder if necessary 👍️